### DISCUSSION TITLE
Bunch of feature requests

### DISCUSSION THREAD

**Discussion by martpie** [avitorio/outstatic#255] [Ideas]
I recently started to play with Outstatic, and I have to say I love its simplicity.

That said, there are a couple of features I am missing, that could be really powerful. I understand not all may make sense or be in the scope of what outstatic is trying to be (or may bring UX challenges), but I'll just share those and you decide what makes sense or not anyway.

moaaar custom fields:

- Reorder custom fields (just for my OCD)
- Image custom field
- Rich Text custom field
- [powerful, but potentially complex] Repeater/List custom field: for example, list of images

Alternative to collections:

- Singleton(s?): One singleton with a name would hold custom fields, for example, put the website title + headline there

Happy to discuss details if you think some are good ideas.

**Comment by avitorio** [comment]
@martpie solid list of suggestions! Thanks!

- Reorder custom fields:
I really like this one and I don't think it would be too difficult to implement. 
Custom fields are basically stored in a json file, so rearranging objects should be straight forward.

- Image custom field (Already added internally and will ship in the next version)
- Rich text custom field: This one sounds tricky since custom field is metadata inside the md/mdx file. I'd have to investigate if that's possible and the best way to achieve it.
- Repeater fields and field groups is something I want to add I just haven't had the time to get it working.
- Singletons is also something I've been meaning to work on for a while and something I think would be easy to get up and running fairly quick, I just haven't prioritised it.

Let me know if you'd like to work on any of them. 
I think custom fields reordering would be a good starting point if you want to help.
I'll be pushing a lot of changes in the next week or so, so I think it would be wise to wait to start on it. I can ping you when the new version is ready!
